CHANGELOG/1.2-CHANGELOG.md

```diff
@@ -0,0 +1,424 @@
+# v1.2 Changelog
```

Comment:
Mind making the corresponding version bumps throughout so this PR can be the one that triggers the release? (x-ref https://github.com/kubernetes-sigs/gateway-api/pull/3062/files)

Reply:
The RELEASE.md says that usually we get this in first though?